test/cql-pytest/test_using_timestamp.py

```diff
@@ -90,3 +90,79 @@ def test_key_writetime(cql, table1):
         cql.execute(f'SELECT writetime(k) FROM {table1}')
     with pytest.raises(InvalidRequest, match='PRIMARY KEY part k'):
         cql.execute(f'SELECT ttl(k) FROM {table1}')
+
+# Reproducer for https://github.com/scylladb/scylladb/issues/14182
+def test_rewrite_using_same_timestamp(cql, table1):
+    table = table1
+    k = unique_key_int()
+    ts = 1000
+    ttl1 = 1
+    ttl2 = 1
+    values = [[1, 1], [1, 2], [2, 1]]
+    errors = []
+    for i in range(3):
+        v1, v2 = values[i]
+
+        # sync with wall-clock on exact second so that expiration won't cross the whole-second boundary
+        t1 = time.time()
+        while t1 - int(t1) >= 0.9:
+            time.sleep(1 - (t1 - int(t1)))
+            t1 = time.time()
+
+        cql.execute(f"INSERT INTO {table} (k, v) VALUES ({k}, {v1}) USING TIMESTAMP {ts} AND TTL {ttl1}")
+        # rewriting right away, so that both cells will carry the same expiry and ttl should return the larger value
```

Comment:
I wouldn't test this case at all. The purpose of the test was supposed to be to check the case where the expiry is NOT the same, and you can achieve this by setting different TTL and not by weird sleeps.
I guess you can also do *this* check but I would make it a separate test (if it exists at all).

Reply:
Separated into a different test case in v4